File: internal/models/process.go
package models

import "time"

type Process struct {
	ID               int32     `db:"id" json:"id"`
	Name             string    `db:"name" json:"name"`
	Description      string    `db:"description" json:"description"`
	Path             string    `db:"path" json:"-"`
	Env              string    `db:"env" json:"env"`
	Status           string    `db:"status" json:"status"`
	ExecuteEverySecs int       `db:"execute_every_secs" json:"execute_every_secs"`
	Running          bool      `db:"running" json:"running"`
	CreatedAt        time.Time `db:"created_at" json:"created_at"`
}

File: client/src/api/process.ts
import { api } from ".";
import { Process } from "../entities";

export const ProcessApi = {
  list: () => api.get<Process[]>("process").json(),

  get: (id: string) => api.get<Process>(`process/${id}`).json(),

  create: (data: Process, file: File) => {
    const formData = new FormData();
    formData.append("file", file);
    formData.append("name", data.name);
    formData.append("description", data.description);
    formData.append("env", data.env);
    formData.append("execute_every_secs", data.execute_every_secs.toString());

    return api.post<Process>("process", { body: formData }).json();
  },
};
